src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUrlBuilder.cs

```diff
@@ -137,6 +138,59 @@ private string ResolveMatchedPath(string matchedPath)
             return RequestPathBase.Add(relativePath).ToString();
         }
 
+        private class PathComparer : IComparer<string>
+        {
+            public int Compare(string x, string y)
+            {
+                // < 0 = x < y
+                // > 0 = x > y
+
+                if (x.Equals(y))
```

Comment:
Talked to @DamianEdwards - we can't use CurrentCulture for sure. Between the two, Ordinal is more correct. But perhaps ultimately we need some kind of custom comparer.